src/http-response-message.js

```diff
@@ -7,14 +8,27 @@ export class HttpResponseMessage {
     this.response = xhr.response;
     this.isSuccess = xhr.status >= 200 && xhr.status < 400;
     this.statusText = xhr.statusText;
-    this.responseType = responseType;
     this.reviver = reviver;
+    this.mimeType = null;
 
     if(xhr.getAllResponseHeaders){
-      this.headers = Headers.parse(xhr.getAllResponseHeaders());
+      try{
+        this.headers = Headers.parse(xhr.getAllResponseHeaders());
+      }catch(err){
+        //if this fails it means the xhr was a mock object so the `requestHeaders` property should be used
+        if(xhr.requestHeaders) this.headers = { headers:xhr.requestHeaders };
+      }
     }else {
       this.headers = new Headers();
     }
+
+    var contentType;
+    if(this.headers && this.headers.headers) contentType = this.headers.headers["Content-Type"];
+    if(contentType) {
+      this.mimeType = responseType = contentType.split(";")[0];
```

Comment:
when you get a change to review the code. i wonder if at the end here there should be a trim() `contentType.split(";")[0].trim()`
just in case the mime type inside the content-type has a trailing space before the semicolon. ie : `['Content-Type'] = "text/html   ; charset=utf-8"`
But i didn't think that would ever be the case. but maybe it should be added to be safe.